### ISSUE TITLE
ValueError: fp16 mixed precision requires a GPU & non-zero exit status 1

### ISSUE THREAD

**Issue by Pico52** [bmaltais/kohya_ss#130]
Traceback (most recent call last):
  File "D:\Projects\Kohya\kohya_ss\train_db.py", line 340, in <module>
    train(args)
  File "D:\Projects\Kohya\kohya_ss\train_db.py", line 57, in train
    accelerator, unwrap_model = train_util.prepare_accelerator(args)
  File "D:\Projects\Kohya\kohya_ss\library\train_util.py", line 1483, in prepare_accelerator
    accelerator = Accelerator(gradient_accumulation_steps=args.gradient_accumulation_steps, mixed_precision=args.mixed_precision,
  File "D:\Projects\Kohya\kohya_ss\venv\lib\site-packages\accelerate\accelerator.py", line 355, in __init__
    raise ValueError(err.format(mode="fp16", requirement="a GPU"))
ValueError: fp16 mixed precision requires a GPU
Traceback (most recent call last):
  File "C:\Users\kupo5\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\kupo5\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "D:\Projects\Kohya\kohya_ss\venv\Scripts\accelerate.exe\__main__.py", line 7, in <module>
  File "D:\Projects\Kohya\kohya_ss\venv\lib\site-packages\accelerate\commands\accelerate_cli.py", line 45, in main
    args.func(args)
  File "D:\Projects\Kohya\kohya_ss\venv\lib\site-packages\accelerate\commands\launch.py", line 1104, in launch_command
    simple_launcher(args)
  File "D:\Projects\Kohya\kohya_ss\venv\lib\site-packages\accelerate\commands\launch.py", line 567, in simple_launcher
    raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
subprocess.CalledProcessError: Command '['D:\\Projects\\Kohya\\kohya_ss\\venv\\Scripts\\python.exe', 'train_db.py', '--pretrained_model_name_or_path=D:/Projects/Stable Diffusion/stable-diffusion-webui/models/Stable-diffusion/uberRealisticPornMerge_urpmv12.safetensors', '--train_data_dir=D:/Projects/Kohya/ProcessLoRA/Current/img', '--resolution=512,512', '--output_dir=D:/Projects/Kohya/ProcessLoRA/Current/model', '--logging_dir=D:/Projects/Kohya/ProcessLoRA/Current/log', '--save_model_as=safetensors', '--output_name=Known-S', '--learning_rate=1e-5', '--lr_scheduler=cosine', '--lr_warmup_steps=200', '--train_batch_size=1', '--max_train_steps=2000', '--save_every_n_epochs=1', '--mixed_precision=fp16', '--save_precision=fp16', '--seed=1234', '--cache_latents', '--bucket_reso_steps=64', '--xformers']' returned non-zero exit status 1.

Using a 1080ti.  I disabled 8Adam and selected f16 as was indicated for others using a 1080ti but it isn't recognizing the GPU?  Did I install something wrong?  I definitely didn't select to use the CPU.  Second issue is related to #124 .

**Comment by bmaltais** [comment]
Try re-running the accelerate config steps. Might just be a missing config option in there.

**Comment by Nezara** [comment]
I had a similar issue. 

Are you running a laptop by chance? if so I think I found the solution.
if you re-run the accelerate config it will ask you "What GPU's should be used for training on this machine?"

Instead of typing in "all" as it prompts you, open up your task manager and look at the performance tab, type in the number that corresponds with the GPU you want to use. eg "1"

When you select "All" its tried to utilize your integrated GPU as well as your dedicated GPU.  Integrated GPU (GPU "0") is not compatible with LORA and this causes it to flag that GPU requirement.

**Comment by RahmelJackson** [comment]
> Are you running a laptop by chance? if so I think I found the solution. if you re-run the accelerate config it will ask you "What GPU's should be used for training on this machine?"

> When you select "All" its tried to utilize your integrated GPU as well as your dedicated GPU. Integrated GPU (GPU "0") is not compatible with LORA and this causes it to flag that GPU requirement.

Would one have to reinstall, where can I change those settings?

**Comment by bmaltais** [comment]
Simply rerun accelerate config and change the values as desired.

**Comment by Pico52** [comment]
I reran the accelerate config and chose "0" as my GPU, as that is my 1080Ti in the task manager.  I am not using a laptop.  I'm getting the same error message.

**Comment by Cauldrath** [comment]
I was getting the same error on a desktop 2080 and I was able to fix it by running the installation steps again.

**Comment by Pico52** [comment]
> I was getting the same error on a desktop 2080 and I was able to fix it by running the installation steps again.

Did you delete the entire repository and start over or just run the original command again?  I tried running it again a while ago and still got the same result, but I didn't delete everything first.

**Comment by Nezara** [comment]
> Are you running a laptop by chance? if so I think I found the solution. if you re-run the accelerate config it will ask you "What GPU's should be used for training on this machine?"
> 
> Instead of typing in "all" as it prompts you, open up your task manager and look at the performance tab, type in the number that corresponds with the GPU you want to use. eg "1"
> 
> When you select "All" its tried to utilize your integrated GPU as well as your dedicated GPU. Integrated GPU (GPU "0") is not compatible with LORA and this causes it to flag that GPU requirement.

I can confirm this DOES NOT solve the issue, also confirmed to not work by others on the aitrepreneur discord.

I cant confirm which step fixes this, but here are some of the steps I took that ended up resolving it.

1. Before setting up the config to use a previously downloaded model, use the option to download a model from scratch. This option doe not just download a model, but also downloads some other dependencies. This seems to help with compatibility with using previously downloaded models. 

2. re-run "Accelerate Config" and select "All" instead of the number representing your graphics card. Sometimes the initial install process is funky/slow or seems to hang.

3. Try running on the low GPU settings first, if this appears to work, swap back to whatever is optimal for you.

4. ensure your folder structures are correct. you should have folders in this appearance. 
Character/image/100_character_name <- Select "image" as your input folder. 
Character/model <- select model as your output folder.
Character/log <- select log as your log folder.

**Comment by Pico52** [comment]
So I got it working, but I did a lot of steps without keeping track of each individual thing.  I think the biggest factor was pretty much spamming the upgrade file a couple of times until it finally I guess downloaded every single version of what it needed.  Seems excessive, and I'm not sure if any of the above solutions were a part of why it worked because I also re-downloaded the models and re-ran accelerate config and am running on low GPU settings.

I guess if someone has this issue, just.. do everything multiple times and hope for the best.
